File: andriodGL/app/src/main/java/z3d/display/particle/ctrl/BaseAnim.java
package z3d.display.particle.ctrl;

import java.util.List;

import z3d.base.Scene_data;

public class BaseAnim {
    public float baseNum;
    public float  num;
    public float  time;
    public  float speed;
    public float  aSpeed;
    public float  beginTime;
    public  float lastTime;
    public  float baseTime;
    public boolean isDeath;
    protected boolean _isActiva;
    protected boolean _isDeath;

    public void updata(float time) {
    }

    public void reset() {
        this._isActiva = false;
        this._isDeath = false;
        this.time = 0;
        this.num = 0;
    }
    public void dataByte(List va, List arr) {
        this.beginTime =(float) arr.get(0);
        float indx001=(float)  arr.get(1);
        if (indx001== -1) {
            this.lastTime = Scene_data.MAX_NUMBER;
        } else {
            this.lastTime =indx001;
        }
    }
}

File: listios/StorageStarted/pan3d/display/particle/ctrl/BaseAnim.m
//
//  BaseAnim.m
//  iosgl
//
//  Created by zhao on 31/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "BaseAnim.h"

@implementation BaseAnim
- (instancetype)init
{
    self = [super init];
    if (self) {
        
    }
    return self;
}
-(void)reset;
{
    self.isActiva = NO;
         self.isDeath = NO;
 
         self.time = 0;
         self.num = 0;
}
-(void)update:(float)t;
{
    BaseAnim* this=self;
 if (this.isDeath) {
     return;
 }
 this.time = t - this.baseTime;
 if (this.isActiva) {
     this.time = this.time - this.beginTime;
     if (this.time > this.lastTime) {
         this.time = this.lastTime - this.beginTime;
         this.isDeath = YES;
     }
     [this coreCalculate];
 } else {
     if (this.time >= this.beginTime) {
         if (this.time >= this.lastTime) {
             this.time = this.lastTime - this.beginTime;
               [this coreCalculate];
             this.isDeath = YES;
         } else {
             this.time = this.time - this.beginTime;
              [this coreCalculate];
         }
         this.isActiva = YES;
     }
 }
}
-(void)coreCalculate;
{
     BaseAnim* this=self;
      this.num = this.speed * this.time + this.aSpeed * this.time * this.time + this.baseNum;
}
 -(void)dataByte:(NSArray<NSNumber*>*)va arr:(NSArray<NSNumber*>*)arr;
{
    
}
@end
